proposals/0298-Processing-of-unknown-enum-values-by-SDL-and-Mobile-libraries.md

```diff
@@ -145,80 +151,18 @@ If a parameter includes an unknown enum value, SDL Core has to cut off such valu
-
-  The validation sequence:
-
-  1. Mobile application sends request to SDL Core through the proxy.
-  2. The proxy checks the request parameters:
-     * If there are no enum values unknown to SDL, the proxy forwards the request to SDL Core without changes.
-
-     * If some of the enum values are unknown to SDL Core the proxy should:
-          * Filter out all RPCs' unsupported enum values including nested RPCs if those are present, before making a request to make sure it is compliant with the current spec
-          * Log a warning level with «unsupported_values» (enum’s/struct’s names)
-
-  **Note**: For `RegisterAppInterface` to filter we should use version of PROTOCOL, but not the MOBILE_API according to system’s flow. **iOS** should be extended with similar mechanism in order to encapsulate filtering logic.
-
-  **For example** :
-  If the UI app requested unsupported `AppHMIType` and version which are not compatible, then the proxy must filter unsupported HMI types.
-
-   - Before filtering :
-       ```
-       "AppHMIType" : ["DEFAULT", "MEDIA", "UKNOWN"]
-       ```
-
-   - After filtering :
-       ```
-       AppHMIType" : ["DEFAULT", "MEDIA"]
-       ```
-
-  ***Note**: In case no HMI types are left after filtering, Mobile Library should throw an exception, otherwise `RegisterAppInterface` RPC will be sent with remaining HMI types.*
-
-  ***Note**: Now system flow is the following: if `RegisterAppInterface` fails with an error then Mobile library should pass the error status to the Mobile Library adding failure reason.*
-
-  ***Note**: The protocol version control logic is partly implemented in Android and not implemented in iOS. However, what is implemented has its flaws and should be updated following the logic above.*
-
-
 ## Potential downsides
-  In the case, new RPCs would added, the new structures/enums should be implemented with additional implementation of filtering mechanism for particular objects.
 
-  **Note:** For `RegisterAppInterface` we cannot keep connection if invalid param is present due to the current flow of the older SDL versions. The connection can be dropped (End Session) by SDL Core but not by mobile device (SDL library).
+There is the case when a Mobile Application has the older API version (for example 6.0) then SDL Core and the SDL Core has the newer API version (for example 7.0).
 
-## Impact on existing code
-  **Android/iOS**
+In the Mobile Application, some parameter is not mandatory and in the SDL Core, the same parameter is become as mandatory. Should keep in mind that the implementation of the SDL Core is updated according to the last API version. And in the described case if the Mobile Application will not send the parameter, that could bring the SDL Core to the incorrect behavior.
```

Comment:
```suggestion
In the Mobile Application, some parameters are not mandatory and in SDL Core, the same parameter has become mandatory. One should keep in mind that the implementation of SDL Core is updated according to the last Mobile API version. And in the described case, if the Mobile Application will not send the parameter, that could bring SDL Core to the incorrect behavior.
```

Reply:
Updated in 3e4835a64afd1b1977009b77d4907a68f37120ee